app/controllers/shelters_controller.rb

```diff
@@ -22,8 +30,13 @@ def edit
 
   def update
     @shelter = Shelter.find(params[:id])
-    @shelter.update(shelter_params)
-    redirect_to "/shelters/#{@shelter.id}"
+    if @shelter.update(shelter_params)
+      redirect_to "/shelters/#{@shelter.id}"
+    else
+      quantity = empty_fields(params).count
```

Comment:
I really like the way you made this work :)

Reply:
Woo! I see lots of opportunities to refactor; however, I recognize we have other priorities right now :)